route/build/src/lake_route.f90

```diff
@@ -0,0 +1,137 @@
+   end do
+  endif
+
+  ! print*, "inside lake, RCHFLX_out(iens,segIndex)%basinprecip = ", RCHFLX_out(iens,segIndex)%basinprecip
+  ! print*, "inside lake, RCHFLX_out(iens,segIndex)%basinevapo = ", RCHFLX_out(iens,segIndex)%basinevapo
+  ! print*, "RPARAM_in(segIndex)%RATECVA", RPARAM_in(segIndex)%RATECVA
+  ! print*, "RPARAM_in(segIndex)%RATECVB", RPARAM_in(segIndex)%RATECVB
+  
+
+  ! perform lake routing based on a fixed storage discharge relationship Q=kS
+  ! no runoff input is added to the lake; the input are only precipitation and evaporation to the lake
+
+  print*, '------lake-simulation-------- '
+  print*, 'node id that is lake ...... = ', NETOPO_in(segIndex)%REACHID ! to check the reach id of lake
+  print*, 'lake param RATECVA ........ = ', RPARAM_in(segIndex)%RATECVA
+  print*, 'lake param RATECVB ........ = ', RPARAM_in(segIndex)%RATECVB
+  print*, 'volume before simulation m3 = ', RCHFLX_out(iens,segIndex)%REACH_VOL(0)
+  print*, 'upstream streamflow m3/s .. = ', RCHFLX_out(iens,segIndex)%REACH_Q_IRF
+  print*, 'upstream precipitation m3/s = ', RCHFLX_out(iens,segIndex)%basinprecip
+  print*, 'upstream evaporation m3/s . = ', RCHFLX_out(iens,segIndex)%basinevapo
+  
+  RCHFLX_out(iens,segIndex)%REACH_VOL(1) = RCHFLX_out(iens,segIndex)%REACH_VOL(0) ! updating storage for current time
+  RCHFLX_out(iens,segIndex)%REACH_VOL(1) = RCHFLX_out(iens,segIndex)%REACH_VOL(1) + q_upstream * dt  ! input upstream discharge from m3/s to m3
+  RCHFLX_out(iens,segIndex)%REACH_VOL(1) = RCHFLX_out(iens,segIndex)%REACH_VOL(1) + RCHFLX_out(iens,segIndex)%basinprecip * dt ! input lake precipitation
+  RCHFLX_out(iens,segIndex)%REACH_VOL(1) = RCHFLX_out(iens,segIndex)%REACH_VOL(1) - RCHFLX_out(iens,segIndex)%basinevapo * dt ! output lake evaporaiton
+  if (RCHFLX_out(iens,segIndex)%REACH_VOL(1) .LT. 0) then; ! set the lake volume as 0 if it goes negative actual evaporation is not calculated here in case low storage mean low evaporaiton...
+    RCHFLX_out(iens,segIndex)%REACH_VOL(1)=0
+  endif
+  ! A has the dimnesion of 1/s; the flux leaving per second
+  RCHFLX_out(iens,segIndex)%REACH_Q_IRF = RPARAM_in(segIndex)%RATECVA * (RCHFLX_out(iens,segIndex)%REACH_VOL(1) ** RPARAM_in(segIndex)%RATECVB)! simplified level pool liner reservoir Q=AS^B
+  RCHFLX_out(iens,segIndex)%REACH_Q_IRF = (min(RCHFLX_out(iens,segIndex)%REACH_Q_IRF * dt, RCHFLX_out(iens,segIndex)%REACH_VOL(1)) )/dt! in case is the output volume is more than lake volume
+  RCHFLX_out(iens,segIndex)%REACH_VOL(1) = RCHFLX_out(iens,segIndex)%REACH_VOL(1) - RCHFLX_out(iens,segIndex)%REACH_Q_IRF * dt ! updating the storage
+  if (RCHFLX_out(iens,segIndex)%REACH_VOL(1) .LT. 0) then; ! set the lake volume as 0 if it goes negative
+    RCHFLX_out(iens,segIndex)%REACH_VOL(1)=0
+  endif
+
+
+  print*, 'lake simulated output m3/s .= ', RCHFLX_out(iens,segIndex)%REACH_Q_IRF
+  print*, 'volume after simulation m3 .= ', RCHFLX_out(iens,segIndex)%REACH_VOL(1)
```

Comment:
Also use ixDesire for spot-check?

Reply:
I guess I need a if statement here as well. in case ixDesire not missing then... right?

Reply:
Ok,  you could follow this example (think just need if statement) https://github.com/nmizukami/mizuRoute/blob/beff9b99c304942d5077eabfc9e93a5feb647dda/route/build/src/irf_route.f90#L195
`ixDesire` is determined somewhere beginning of the program based on `desireId` in control file (optional input).

Reply:
I am trying to test the changed I make here for `ixDesire` how is the control file should be to account for the desire id... it seems that is not a simple integer read...
```
! MISCELLANEOUS
case('<desireId>'   );          read(cData,*,iostat=io_error) desireId          ! turn off checks or speficy reach ID if 
```
and this is how I put it in the control file...
```
<desireId>          95                              ! turn off checks or speficy reach ID if necessary to print on screen
```

Reply:
at this moment it give such a error reading the control file. I suspect I am not providing the right input...
`FATAL ERROR: init_model/read_control/problem disentangling cLines(iLine) [strin
 g=]          20`

Reply:
Yes, looks control file correct, but based on error message, i think you have empty line in control file?    `desireId` is integer, and this is how it is read.

Reply:
Yes, I had an additional like in control file. now it seems to read the ID but it still returns the -9999 in stead of the if of the `ixDesire`.

Reply:
I think I know why.  The feature/mpi-pio you branched out does not detect ixDesire if you run in one task.     I made some changes in the other pull-request so one task runs works for this.   Do you really need to use now?   if not, you could leave it as is.   Once all are merged, hope it works.   you could try multiple task.

Reply:
actually I realized this as it was not printing the ixDesire in network_topo.f90. and also when a segment is given to it instead of -9999 for seg_outlet is doesn't run... no I don't need it... I can just do a hard coded value for checking the code and add the code to the pull.

Reply:
no actually I think it is not that easy to check this with hard coded values, It will propagate over all the functions and index of ID may not be the same as ID, etc... I can just add this to the pull... commented the if and we can remove the comment later when ixdesire works..